src/util/rocksdb.rs

```diff
@@ -46,56 +46,75 @@ pub fn open_opt(opts: Options,
 }
 
 pub fn new_engine(path: &str, cfs: &[&str]) -> Result<DB, String> {
-    let mut opts = Options::new();
-    opts.enable_statistics();
-    let mut cfs_opts = vec![];
-    for _ in 0..cfs.len() {
-        cfs_opts.push(Options::new());
+    let mut db_opts = Options::new();
+    db_opts.enable_statistics();
+    let mut cfs_opts = HashMap::default();
+    for cf in cfs {
+        cfs_opts.insert(*cf, Options::new());
     }
-    new_engine_opt(opts, path, cfs, cfs_opts)
+    new_engine_opt(path, db_opts, cfs_opts)
 }
 
-fn check_column_families(path: &str,
-                         needed_cfs: &[&str],
-                         cfs_opts: &[&Options])
-                         -> Result<(), String> {
+fn check_and_open(path: &str,
+                  mut db_opt: Options,
+                  cfs_opts: HashMap<&str, Options>)
+                  -> Result<DB, String> {
     // If db not exist, create it.
     if !db_exist(path) {
```

Comment:
There is no same logic can be reused, on the other hand it is more clear separate the conditions db exist and not exist.